services/datamanager/builtin/capture_test.go

```diff
@@ -235,13 +198,126 @@ func TestDataCaptureEnabled(t *testing.T) {
 	}
 }
 
-func waitForCaptureFiles(captureDir string) {
+func TestSwitchResource(t *testing.T) {
+	captureDir := t.TempDir()
+	mockClock := clk.NewMock()
+	// Make mockClock the package level clock used by the dmsvc so that we can simulate time's passage
+	clock = mockClock
+
+	// Set up robot config.
+	config, deps := setupConfig(t, enabledTabularCollectorConfigPath)
+	config.CaptureDisabled = false
+	config.ScheduledSyncDisabled = true
+	config.CaptureDir = captureDir
+
+	// Build and start data manager.
+	dmsvc, r := newTestDataManager(t)
+	defer func() {
+		test.That(t, dmsvc.Close(context.Background()), test.ShouldBeNil)
+	}()
+
+	resources := resourcesFromDeps(t, r, deps)
+	err := dmsvc.Reconfigure(context.Background(), resources, resource.Config{
+		ConvertedAttributes: config,
+	})
+	test.That(t, err, test.ShouldBeNil)
+	passTimeCtx1, cancelPassTime1 := context.WithCancel(context.Background())
+	donePassingTime1 := passTime(passTimeCtx1, mockClock, captureInterval)
+
+	waitForCaptureFilesToExceedN(captureDir, 0)
+	testFilesContainSensorData(t, captureDir, nil)
+
+	cancelPassTime1()
+	<-donePassingTime1
+
+	// Change the resource named arm1 to show that the data manager recognizes that the collector has changed with no other config changes.
+	for resource := range resources {
+		if resource.Name == "arm1" {
+			newResource := inject.NewArm(resource.Name)
+			newResource.EndPositionFunc = func(ctx context.Context, extra map[string]interface{}) (spatialmath.Pose, error) {
+				// Return a different value from the initial arm1 resource.
```

Comment:
What does the original arm resource return?

Reply:
The original arm returns `spatialmath.NewPoseFromPoint(r3.Vector{X: 1, Y: 2, Z: 3}` (code [here](https://github.com/viamrobotics/rdk/blob/ca09119c49ac2e984e1141557a7243202609e8b1/services/datamanager/builtin/builtin_test.go#L43))